department-of-veterans-affairs/caseflow
# Make MetricsService report to DataDog

This PR will allow us to send metrics in parallel to Prometheus and DataDog. We can test out DataDog before turning off the equivalent functionality in Prometheus.

In order for this to work, we need to update `appeals-deployment` to add the API key to the servers: https://github.com/department-of-veterans-affairs/appeals-deployment/pull/1115.

See also: https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/776.

## Test
See this in action on [UpDataDog](https://app.datadoghq.com/notebook/22687/Nick%20Test).

![image](https://user-images.githubusercontent.com/25331532/33967310-e5dfba3a-e030-11e7-95a8-6cfc1ff37688.png)

I tested this by:

1. Following the [external deps](https://github.com/department-of-veterans-affairs/caseflow/#running-caseflow-connected-to-external-depedencies) steps.
1. Set DATADOG_API_KEY to `credstash -t appeals-credstash get datadog.api_key`
1. `rails c -e staging`
1. Make an external dep request, like:
    ```rb
    AppealRepository.appeals_by_vbms_id('any value')  
    ```
1. Wait a few moments.
1. See your data point show up in DataDog.

I also added changes to track DB connections but to be honest I'm not sure how to test that.
